poh/src/poh_recorder.rs

```diff
@@ -1006,6 +1024,57 @@ impl SharedWorkingBank {
     }
 }
 
+/// Wrapper around a atomic-u64 that prevents modifying outside
+/// of `PohRecorder`.
+#[derive(Clone)]
+pub struct SharedTickHeight(Arc<AtomicU64>);
+
+impl SharedTickHeight {
+    pub fn load(&self) -> u64 {
+        self.0.load(Ordering::Acquire)
+    }
+
+    fn new(tick_height: u64) -> Self {
+        Self(Arc::new(AtomicU64::new(tick_height)))
+    }
+
+    fn store(&self, tick_height: u64) {
+        self.0.store(tick_height, Ordering::Release);
+    }
+
+    fn increment(&self) {
+        self.0.fetch_add(1, Ordering::Release);
+    }
+}
+
+/// Wrapper around a atomic-u64 that may be None.
+// Uses the flag of u64::MAX to indicate None; this does not
+// need to be observable outside of PohRecorder.
+#[derive(Clone)]
+pub struct SharedLeaderFirstTickHeight(Arc<AtomicU64>);
```

Comment:
Thought about having the same struct as above, but the loading is different since this can be None (via u64::MAX flag).

Thought cleaner to just separate them, even though they are very similar. Will also make it a bit harder to confuse the two once the shared structs are cloned outside of PohRecorder.